drivers/flash/flash_mcux_flexspi_hyperflash.c

```diff
@@ -395,6 +395,14 @@ static int flash_flexspi_hyperflash_read(const struct device *dev, off_t offset,
 {
 	struct flash_flexspi_hyperflash_data *data = dev->data;
 
+	if (len == 0) {
+		return 0;
+	}
+
+	if (!buffer) {
```

Comment:
Not compliant with Coding Guidelines, Zephyr rule 85 (https://gitlab.com/MISRA/MISRA-C/MISRA-C-2012/Example-Suite/-/blob/master/R_14_04.c), should be `buffer == NULL`.

Reply:
OK, now I see that the entire driver keeps that style, so the decision is on you.